### PR TITLE
change cable.coffee to cable.js [ci skip]

### PR DESCRIPTION
**Pull request by y-yagi**
In #23935, cable file was to be provided by the javascript instead of coffeescript,
doc was also been modified to use javascript.

### CODE REVIEW

File: guides/source/action_cable_overview.md [hunk 1]
```diff
@@ -142,12 +142,18 @@ established using the following Javascript, which is generated by default in Rai
...
+//= require_tree ./channels
+
+(function() {
+  this.App || (this.App = {});
+
+  App.cable = ActionCable.createConsumer();
 
-@App = {}
-App.cable = ActionCable.createConsumer()
+}).call(this);
```
Comment (vipulnsward) [inline guides/source/action_cable_overview.md]: Is this needed?

Comment (y-yagi) [inline guides/source/action_cable_overview.md]: This is necessary in order to use `this` in strict mode(If I am not mistaken).  
In addition, CoffeeScript is to use `call (this)` to convert the immediate function, have to match it.

**Comment by maclover7** [comment]
Looks like on master right now, we have `cable.js` be a JS file, but then we have all of the channels be CS files... I think we should have a configuration option for having everything either be JS or CS.

https://github.com/rails/rails/blob/master/actioncable/lib/rails/generators/channel/channel_generator.rb#L17

File: actioncable/README.md [hunk 2]
```diff
@@ -86,12 +86,18 @@ end
...
+```js
+// app/assets/javascripts/cable.js
+//= require action_cable
+//= require_self
+//= require_tree ./channels
+
+(function() {
+  this.App || (this.App = {});
+
+  App.cable = ActionCable.createConsumer("ws://cable.example.com");
```
Comment (vipulnsward) [inline actioncable/README.md]: Extra newline after this  ✂️

File: guides/source/action_cable_overview.md [hunk 3]
```diff
@@ -142,12 +142,18 @@ established using the following Javascript, which is generated by default in Rai
...
+```js
+// app/assets/javascripts/cable.js
+//= require action_cable
+//= require_self
+//= require_tree ./channels
+
+(function() {
+  this.App || (this.App = {});
+
+  App.cable = ActionCable.createConsumer();
```
Comment (vipulnsward) [inline guides/source/action_cable_overview.md]: ✂️

**Comment by kaspth** [comment]
Merged @ 017c7f4. Thanks!

**Comment by kaspth** [comment]
@maclover7 I think there was some discussion around inferring CS/JS from whether or not users depended on `coffee-rails` 😁

**Comment by vipulnsward** [comment]
Yes Prathamesh is working on it.

On Saturday 7 May 2016, Kasper Timm Hansen <notifications@github.com> wrote:

> @maclover7 <https://github.com/maclover7> I think there was some
> discussion around inferring CS/JS from whether or not users depended on
> coffee-rails 😁
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/rails/rails/pull/24901#issuecomment-217662733>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAipSt24-GRV7TnUS5uKHCpj9hXUecKbks5p_OVOgaJpZM4IZVv8>
> .
>


-- 
Vipul A.M.
+91-8149-204995

**Comment by prathamesh-sonpatki** [comment]
This is the PR for inferring which javascript engine user is using - https://github.com/rails/rails/pull/24918

**Pull request closed without merging**